### PR TITLE
Feature/OB-30_FooterLayout

### PR DESCRIPTION
**Pull request by ain-odin**
1. Added new `footerLists.js` api in `pages/api` Снимок экрана 2023-08-31 в 17 31 57

2.  Edited (renamed and new data) `barbershops.js` api -> `barbershopsInfo.js` Снимок экрана 2023-08-31 в 15 17 08

3. Added new component `FooterNavList` in `src/components` Снимок экрана 2023-08-31 в 15 30 58

4. Added `Footer` component in `src/components` using `FooterNavList`
5. Created tests for new components
6. Added `Footer` component to the page fetching data for several elements Снимок экрана 2023-08-31 в 17 32 17

7. This is how it looks on the `page` Снимок экрана 2023-08-31 в 14 42 43 Снимок экрана 2023-08-31 в 14 42 54

### CODE REVIEW

File: pages/api/barbershopsInfo.js
Comment (Alex-2kZharkov) [inline pages/api/barbershopsInfo.js:1]: please, change the .js extension to .ts . We still in the TypeScript environment even if you are not using any types

File: pages/api/footerLists.js
Comment (Alex-2kZharkov) [inline pages/api/footerLists.js:1]: Same about extension

File: public/images/bg/white-grunge-180.svg
Comment (Alex-2kZharkov) [inline public/images/bg/white-grunge-180.svg:1]: We already have such an image. let's reuse it and do not duplicate ourselves. Check `public/images/bg` folder. Maybe you can just rotate the image with `transform` property to reuse it

File: src/app/page.tsx [hunk 1]
```diff
@@ -3,10 +3,20 @@
...
 import { PromoCard } from '@/components/PromoCard';
 
+import { Footer } from '@/components/Footer';
+
 const Home = async () => {
-  const response = await fetch('http://localhost:3000/api/barbershops');
-  const jsonResponse = await response.json();
-  const numberOfBarbershops = jsonResponse.data;
+  const barbershopsInfoResponse = await fetch('http://localhost:3000/api/barbershopsInfo');
+  const barbershopsInfoJsonResponse = await barbershopsInfoResponse.json();
```
Comment (Alex-2kZharkov) [inline src/app/page.tsx]: let's do it more convinient: `const {branchesTotal, countriesTotal, capitalsTotal} = await barbershopsInfoResponse.json();`

numberOfSomething - I find this naming too long and not informative, let's stick to original version

File: src/app/page.tsx [hunk 2]
```diff
@@ -3,10 +3,20 @@
...
+  const numberOfBranches = barbershopsInfoJsonResponse.branchesTotal;
+  const numberOfCountries = barbershopsInfoJsonResponse.countriesTotal;
+  const numberOfCapitals = barbershopsInfoJsonResponse.capitalsTotal;
+
+  const footerListsResponse = await fetch('http://localhost:3000/api/footerLists');
+  const footerListsJsonResponse = await footerListsResponse.json();
+  const { footerMainList } = footerListsJsonResponse;
```
Comment (Alex-2kZharkov) [inline src/app/page.tsx]: check the comment above about destructuring.

Plus, I have one more improvement! Use Promise.all() for these 2 request. They do not depend on each other, so you can do it

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all#description

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import { FooterNavList } from '../FooterNavList';
+
+interface FooterProps {
+  branchesTotal: number;
+  countriesTotal: number;
+  capitalsTotal: number;
+  footerMainTitle: string;
+  footerServicesTitle: string;
+  footerInfoTitle: string;
+  footerMainList: any;
```
Comment (Alex-2kZharkov) [inline src/components/Footer/Footer.tsx]: any is forbidden in 90+ percent of usages. I think it is an array of something, check the API and create the required types

File: src/components/Footer/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        <div className="footer__menues flex justify-between basis-footer-basis py-10 px-5 mb-10 border-y-2 border-y-solid border-y-orange-title">
+          <FooterNavList navTitle={footerMainTitle} navList={footerMainList} />
+          <FooterNavList navTitle={footerServicesTitle} navList={footerServicesList} />
+          <FooterNavList navTitle={footerInfoTitle} navList={footerInfoList} />
+        </div>
+      </div>
+      <div className="footer__copyrights font-button_font text-footer-grey text-xs font-semibold text-right uppercase pr-10">
+        <p>
+          <span className="footer__copyrights-stress text-white text-[10px] font-bold">
+            Копирование материалов запрещено 18+{' '}
```
Comment (Alex-2kZharkov) [inline src/components/Footer/Footer.tsx]: You sure you need the `{' '}` part? it is not convenient

File: src/components/Footer/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+              <p className="footer__text text-footer-grey font-button_font text-sm leading-normal mb-[0.5em]">
+                OldBoy Barbershop — намного больше, чем просто мужская парикмахерская, это даже
+                больше, чем мужской клуб по интересам, это именно то место, где ты найдёшь себя и
+                свой стиль.
+              </p>
+            </div>
+          </div>
+        </div>
+        <div className="footer__menues flex justify-between basis-footer-basis py-10 px-5 mb-10 border-y-2 border-y-solid border-y-orange-title">
+          <FooterNavList navTitle={footerMainTitle} navList={footerMainList} />
```
Comment (Alex-2kZharkov) [inline src/components/Footer/Footer.tsx:80]: Yes, very clever!

File: src/components/FooterNavList/FooterNavList.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+export const FooterNavList = ({ navTitle, navList }: FooterNavListProps): ReactElement => {
+  return (
+    <div>
+      <h3 className="text-footer-grey font-rex text-[1.6rem] leading-loose text-left uppercase pt-4">
+        {navTitle}
+      </h3>
+      <ul className="text-left pb-4">
+        {navList.map(({ title, href }: any) => {
```
Comment (Alex-2kZharkov) [inline src/components/FooterNavList/FooterNavList.tsx]: fix the any here as well

**Review by Alex-2kZharkov — CHANGES_REQUESTED** [review]

File: src/app/page.tsx (hunk 2 above)
Comment (Alex-2kZharkov) [reply src/app/page.tsx]: ok, we will discuss it during the stand up

File: src/components/FooterNavList/FooterNavList.tsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import { ReactElement } from 'react';
+import Link from 'next/link';
+
+interface FooterNavListProps {
+  navTitle: string;
+  navList: Array<{ title: string; href: string }>;
```
Comment (Alex-2kZharkov) [inline src/components/FooterNavList/FooterNavList.tsx]: I think you need a separate type like LinkOptions, place it in the `src/types/link.ts` file. You can reuse it in the Footer component as well

File: pages/api/barbershopsInfo.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
+const handler = (
+  req: any,
+  res: {
+    status: (arg0: number) => {
```
Comment (Alex-2kZharkov) [inline pages/api/barbershopsInfo.ts]: looks not so good yet, we will improve it together

**Review by Alex-2kZharkov — CHANGES_REQUESTED** [review]

File: public/images/bg/white-grunge-180.svg
Comment (Alex-2kZharkov) [inline public/images/bg/white-grunge-180.svg:1]: Don't forget to delete this image

**Review by Alex-2kZharkov — CHANGES_REQUESTED** [review]

File: src/components/FooterNavList/FooterNavList.tsx (hunk 7 above)
Comment (Alex-2kZharkov) [reply src/components/FooterNavList/FooterNavList.tsx]: @ain-odin don't forget about this one🙂

Comment (ain-odin) [reply src/components/FooterNavList/FooterNavList.tsx]: Screenshot 2023-09-02 at 12 13 34

File: pages/api/barbershopsInfo.ts (hunk 8 above)
Comment (ain-odin) [reply pages/api/barbershopsInfo.ts]: Screenshot 2023-09-02 at 12 13 20
Screenshot 2023-09-02 at 12 13 05

Comment (Alex-2kZharkov) [reply pages/api/barbershopsInfo.ts]: excellent!🏆

**Review by Alex-2kZharkov — APPROVED** [review]
Good job! 🎯

**Pull request merged**